service/src/main/java/org/dice_research/fc/paths/imprt/PredicateRetriever.java

```diff
@@ -0,0 +1,95 @@
+package org.dice_research.fc.paths.imprt;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Predicate;
+import org.aksw.jena_sparql_api.core.QueryExecutionFactory;
+import org.apache.jena.query.QueryExecution;
+import org.apache.jena.query.QuerySolution;
+import org.apache.jena.query.ResultSet;
+import org.dice_research.fc.sparql.filter.NamespaceFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Retrieves all or a select few predicates from a KG.
+ * 
+ * @author Alexandra Silva
+ *
+ */
+public class PredicateRetriever {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PredicateRetriever.class);
+
+  /**
+   * The query executioner.
+   */
+  private QueryExecutionFactory qef;
+
+  /**
+   * Predicate variable in queries.
+   */
+  private final String PREDICATE_VARIABLE = "p";
+
+  /**
+   * Namespace filter to apply to queries.
+   */
+  private NamespaceFilter filter;
```

Comment:
I would suggest to have a `Predicate<String> filter`. This gives much more flexibility. You can of course provide a constructor which takes the namespace information and creates the predicate. However, I can imagine that we may want to exclude some specific properties, in the future :thinking: 

Another way to keep it easy might be to simply let NamespaceFilter extend the `Predicate<String>` interface and offer two constructors: one (easier) that takes a NamespaceFilter and a second (more flexible) constructor that takes any `Predicate<String>` implementation.